.github/workflows/build.yml

```diff
@@ -5,42 +5,51 @@ on:
 
 jobs:
   build_wheels:
-    name: Build wheels on ${{ matrix.os }}
+    name: Build wheels on ${{ matrix.os }}-${{ matrix.cibw_archs }}
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         include:
-          - os: ubuntu-latest
-            cibw_archs: auto64
-          - os: ubuntu-latest
+          - os: ubuntu-latest # linux x86_64
+            cibw_archs: x86_64
+            cibw_skip: "pp*"
+          - os: ubuntu-latest # linux arm64 gnu
             cibw_archs: aarch64
+            cibw_skip: "pp* *musllinux*"
+          - os: ubuntu-latest # linux arm64 musl
+            cibw_archs: aarch64
+            cibw_skip: "pp* *manylinux*"
           - os: windows-latest
-            cibw_archs: auto64
-          - os: macos-latest
+            cibw_archs: AMD64 ARM64
+            cibw_skip: "pp*"
+          - os: macos-14
             cibw_archs: universal2
+            cibw_skip: "pp*"
 
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
 
       - name: Set up QEMU
-        if: matrix.cibw_archs == 'aarch64'
-        uses: docker/setup-qemu-action@v2
+        if: runner.os == 'Linux' && matrix.cibw_archs == 'aarch64'
+        uses: docker/setup-qemu-action@v3
         with:
           platforms: arm64
 
-      - name: Build wheels
-        uses: pypa/cibuildwheel@v2.14
+      - name: Build wheels on ${{ matrix.os }}-${{ matrix.cibw_archs }}
+        uses: pypa/cibuildwheel@v2.17
         env:
           CIBW_BUILD_FRONTEND: build
-          CIBW_PROJECT_REQUIRES_PYTHON: ">=3.7"
```

Comment:
Is this going to break python3.7 & 3.8 users?

3.8 seems to have a good amount of users: https://pypistats.org/packages/pycocotools

Reply:
This is because numpy no longer supports versions lower than 3.9 😿.
https://numpy.org/neps/nep-0029-deprecation_policy.html#support-table

The other dependency, matplotlib, also does not support below 3.9. 
https://github.com/matplotlib/matplotlib/blob/d45dfbb245b991e51f60a5083d61d0130b2ecc1b/pyproject.toml#L45

By modifying `python_requires` in setup.py, we can ensure that pycocotools 2.0.7 is selected for Python 3.8 and below. (pip or other dependency manager select the appropriate version.)